x/gov/keeper/deposit.go

```diff
@@ -162,6 +163,75 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// chargeDeposit will charge proposal cancellation fee (deposits * proposal_cancel_burn_rate)  and
+// send to a destAddress if defined or burn otherwise.
+// Remaining funds are send back to the depositor.
+func (keeper Keeper) chargeDeposit(ctx sdk.Context, proposalID uint64, destAddress, proposalCancelRate string) error {
+	store := ctx.KVStore(keeper.storeKey)
+	rate := sdk.MustNewDecFromStr(proposalCancelRate)
+	var cancellationCharges sdk.Coins
+
+	for _, deposits := range keeper.GetDeposits(ctx, proposalID) {
+		depositerAddress := sdk.MustAccAddressFromBech32(deposits.Depositor)
+		var remainingAmount sdk.Coins
+
+		for _, deposit := range deposits.Amount {
+			burnAmount := sdk.NewDecFromInt(deposit.Amount).Mul(rate).TruncateInt()
+			// remainin amount = deposits amount - burn amount
+			remainAmount := sdk.NewCoin(
+				deposit.Denom,
+				deposit.Amount.Sub(burnAmount),
+			)
+			remainingAmount = remainingAmount.Add(remainAmount)
```

Comment:
```suggestion
			depRemainAmount := sdk.NewCoin(
				deposit.Denom,
				deposit.Amount.Sub(burnAmount),
			)
			remainingAmount = remainingAmount.Add(remainAmount)
```

It's confusing to have `remainAmount` and `remainingAmount`. Consider renaming the 1st one to something with `dep`